server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchAction.java

```diff
@@ -179,6 +179,7 @@ private static SearchWsResponse buildResponse(List<ComponentDto> components, Org
       .setOrganization(organization.getKey())
       .setId(dto.uuid())
       .setKey(dto.key())
+      .setProjectKey(dto.projectKey())
```

Comment:
The project key should NOT come from the DTO.
We should : 
1. Get all project uuids from searched components
2. Get all projects dto from project uuids => create a map of dto by project uuid 
3. Use this map to get the project key of a component